File: Go/Day01.go
package main

import AH "./adventhelper"

func part1(vec []int) (acc int) {
	for i := 0; i < len(vec) - 1; i++ {
		if vec[i] < vec[i+1] {
			acc += 1
		}
	}

	return
}

func part2(vec []int) (acc int) {
	for i := 0; i < len(vec) - 3; i++ {
		if vec[i] < vec[i+3] {
			acc += 1
		}		
	}

	return
}

func main() {
	ints, _ := AH.ReadIntFile("../input/input01.txt")

	AH.PrintSoln(1, part1(ints), part2(ints))

	return
}

File: Haskell/Day01.hs
import AdventHelper

part2 :: [Integer] -> [Bool]
part2 x = map (\(l,r) -> l > r) $ zip (drop 3 x) x

main = do
  putStrLn "Day 1"
  f <- readFile "../input/input01.txt"
  let ss = map(read) $ lines f :: [Integer]

  printSoln 1 (length $ filter (< 0) $ diff ss)
  printSoln 2 (length $ filter (== True) $ part2 ss)
